cmd/soroban-cli/src/commands/contract/read.rs

```diff
@@ -94,6 +94,8 @@ pub enum Error {
     // TODO: the Display impl of host errors is pretty user-unfriendly
     //       (it just calls Debug). I think we can do better than that
     Host(#[from] HostError),
+    #[error("no ledger key was found for given contract")]
```

Comment:
"no ledger key" is a little confusing because a developer will have provided a contract data key, and the error sounds like the user has failed to provide one, when in fact no ledger entry can be found that matches the key.

An error that focuses on the fact that no data can be found matching the key would be clearer.

Reply:
Hmm. Yes, that's true. I'll update that tomorrow morning.